docs/motivation-and-use-cases.md

```diff
@@ -0,0 +1,138 @@
+- support for byzantine consensus protocols sounds relevant
+- support for updating many uncoordinated nodes
+
+Thoughts:
+This is a particularly intersting use case as it requires more significantly new and unexplored functionality from cluster than any of the others here.  As @hsanjuan mentioned in his original write up "The key here is to understand what the trust model is in a pinning ring, how members gain and lose trust, and who can take what actions".  On a similar note a byzantine consensus protocol may greatly help keep the ring working smoothly even when some peers misbehave.  This use case also presents challenges regarding how to get nodes to update when they are managed by different individuals.  The current approach to updating requires all nodes to be shut down at the same tiem which may be unrealistic here.
+
+This would be a "dynamic" cluster with lots of peer membership churn.  ipfs-cluster will need to thoroughly work out any issues with peer membership changes for this to work well.  Additionally peer set churn makes things like fault tolerant sharding with potentially expensive data recovery algorithms hard to support as recovery might be run frequently.  My guess is that pinning with high replication factors will be better for this use case.
+
+
+## Big ipfs node for ubuntu archive seeding (reported here https://discuss.ipfs.io/t/ubuntu-archive-on-top-of-ipfs/1579)
+
+Actors:
+	admin of package-seeding cluster
+	mirror site providing package data
+	users fetching packages over ipfs
+
+Description:  An admin wishes to set up an ipfs node storing a mirror of ubuntu deb packages to support an apt transport that downloads deb packages from ipfs.  The admin wishes to download the packages and directory structure from one of the existing mirrors over http and store in ipfs.  The admin does not have a server with the 2TB of storage necessary to host the entire mirror and so cannot fit all packages on a single ipfs node.  However the admin does have access to a set of smaller servers (say 4 servers of 50GB) that together fulfill the total storage capacity of the mirror.  The admin installs ipfs-cluster on each server and then commands the cluster to download the mirror data.  During download the cluster allocates different pieces of the mirror directory to different machines, spreading load evenly.  If there is extra space on the servers then replication of packages is a bonus, but this is not a primary concern for this usecase.  The cluster hosting the mirror can be assumed stable, with a fixed number of servers all ran by a single admin or administrative body.  After packages are added to the cluster users will fetch packages by path name from the root hash (QmAAA.../mirrorDir1/mirrorDir2/package.deb).  
+
+Implied ipfs-cluster requirements
+- ipfs-cluster can handle importing, sharding and distributing across the cluster a file too big for one node, see PR #268.
```

Comment:
To be exact, it's a tree too big for one node. Files are rather small in the apt repository. The problem of distributing a huge file among several peers and the problem of distributing a huge archive of small files among several peers can be approached slightly different, even though we probably fix them the same way (because they're large trees in the end).

Reply:
Got it, thanks for the clarification.